src/UnscentedKalmanInversion.jl

```diff
@@ -73,30 +77,62 @@ function Unscented(
     update_freq::IT = 1,
     modified_unscented_transform::Bool = true,
     prior_mean::Union{AbstractVector{FT}, Nothing} = nothing,
+    N_ens::IT = 2 * size(u0_mean, 1) + 1,
 ) where {FT <: AbstractFloat, IT <: Int}
 
     N_par = size(u0_mean, 1)
     # ensemble size
-    N_ens = 2 * N_par + 1
 
-    c_weights = zeros(FT, N_par)
     mean_weights = zeros(FT, N_ens)
     cov_weights = zeros(FT, N_ens)
 
-    # set parameters λ, α
-    α = min(sqrt(4 / N_par), 1.0)
-    λ = α^2 * N_par - N_par
+    if N_ens == 2 * N_par + 1
+        c_weights = zeros(FT, N_par)
 
+        # set parameters λ, α
+        α = min(sqrt(4 / N_par), 1.0)
+        λ = α^2 * N_par - N_par
 
-    c_weights[1:N_par] .= sqrt(N_par + λ)
-    mean_weights[1] = λ / (N_par + λ)
-    mean_weights[2:N_ens] .= 1 / (2 * (N_par + λ))
-    cov_weights[1] = λ / (N_par + λ) + 1 - α^2 + 2.0
-    cov_weights[2:N_ens] .= 1 / (2 * (N_par + λ))
 
-    if modified_unscented_transform
-        mean_weights[1] = 1.0
-        mean_weights[2:N_ens] .= 0.0
+        c_weights[1:N_par] .= sqrt(N_par + λ)
+        mean_weights[1] = λ / (N_par + λ)
+        mean_weights[2:N_ens] .= 1 / (2 * (N_par + λ))
+        cov_weights[1] = λ / (N_par + λ) + 1 - α^2 + 2.0
+        cov_weights[2:N_ens] .= 1 / (2 * (N_par + λ))
+
+        if modified_unscented_transform
+            mean_weights[1] = 1.0
+            mean_weights[2:N_ens] .= 0.0
+        end
```

Comment:
I think this is shared between the two options, so we can take it out of the if/else statement.

Reply:
Thanks for the review! I have moved this outside the loop and add sigma_points::String, and update the tests